Source/Urho3D/Urho2D/ParticleEmitter2D.cpp

```diff
@@ -262,11 +262,21 @@ void ParticleEmitter2D::UpdateMaterial()
         sourceBatches_[0].material_ = 0;
 }
 
+URHO3D_EVENT(E_PARTICLESEND, ParticlesEnd) {
+}
+URHO3D_EVENT(E_PARTICLESDURATION, ParticlesDuration) {
+}
+
 void ParticleEmitter2D::HandleScenePostUpdate(StringHash eventType, VariantMap& eventData)
 {
     using namespace ScenePostUpdate;
+    unsigned tests = (numParticles_ > 0 ? 1 : 0) | (emissionTime_ > 0.0f ? 2 : 0);

```

Comment:
Please avoid this unneeded esoteric bitmagic, e.g. in this way:

```
bool wasEmitting = emissionTime_ > 0.0f;
bool hasParticles = numParticles_ > 0;
```

PS: And check for trailing spaces